lib/matplotlib/artist.py

```diff
@@ -543,6 +535,10 @@ def set_picker(self, picker):
               artist, return *hit=True* and props is a dictionary of
               properties you want added to the PickEvent attributes.
 
+            - *deprecated*: For `.Line2D` only, *picker* can also be a float
+              that sets the tolerance for checking whether an event occurred
```

Comment:
Hmmm, you can see how this ended up here, given that we can also pass a function at this point.

Reply:
But at least functions are not misinterpretable as numbers...

Reply:
Is the other solution to only set the pick radius if picker is not a `bool`?

Reply:
Sure, you can do that (but what about `np.bool_(True)`?).  But e.g. LineCollections also have a pickradius which is *not* settable with set_picker, and noone ever requested to change that, so...

Reply:
Well, I guess you can deprecate it and see if anyone notices...  I do think there is some advantage to having the pick logic all in one place, rather than here and in `set_pickradius`, but don't feel strongly about it

Reply:
I have some further cleanups too, but as always and as you say, if people show up with pitchforks, we can always revert the change...

Reply:
> I do think there is some advantage to having the pick logic all in one place

I was thinking about t his as well, but it occurred to me that `set_picker(5)` is not quite readable. 
 We can later always try and find something even better, cleaning this up now and aligning withwith other parts of the API is a reasonable step.